### ISSUE TITLE
Update to the latest web3.js

### ISSUE THREAD

**Issue by JirkaChadima** [windingtree/wt-js-libs#243]
After https://github.com/ethereum/web3.js/releases/tag/v1.0.0-beta.38 things stop to work. Because fuck semantic versioning, we can totally change the whole library between two beta releases.

So we need to investigate, update and make our library compatible.

**Comment by JirkaChadima** [comment]
I think this is the original issue: https://github.com/ethereum/web3.js/issues/2266

**Comment by JirkaChadima** [comment]
So, the problem is indeed the incompatibility of [truffle-contract](https://github.com/trufflesuite/truffle/issues/1690) with the latest changes in [web3.js](https://github.com/ethereum/web3.js/issues/2266). We cannot upgrade web3 until truffle-contract/ganache-cli are made compatible with that, because we cannot run the tests without it. An optional path would be to ditch truffle-contract in favour of something else, but solving this seems to be top priority in truffle (https://github.com/trufflesuite/truffle/issues/1690) so I wouldn't bother with that and just wait.

**Comment by JirkaChadima** [comment]
Should get unblocked once https://github.com/ethereum/web3.js/pull/2426 is merged and released

**Comment by nivida** [comment]
I'm currently preparing the [Web3 Module API](https://github.com/ethereum/web3.js/pull/2236) which could probably be interesting for this process.

**Comment by JirkaChadima** [comment]
Closing this in favor of #255
